src/vivarium_gates_iv_iron/components/maternal_bmi.py

```diff
@@ -0,0 +1,84 @@
+import numpy as np
+import pandas as pd
+from vivarium.framework.engine import Builder
+from vivarium.framework.population import SimulantData
+from vivarium.framework.randomness import get_hash
+
+from vivarium_gates_iv_iron.constants import data_values, models
+from vivarium_gates_iv_iron.data import sampling
+
+
+class MaternalBMIExposure:
+
+    @property
+    def name(self):
+        return 'maternal_bmi_exposure'
+
+    def setup(self, builder: Builder):
+        self.randomness = builder.randomness.get_stream(self.name)
+        self.hemoglobin = builder.value.get_value('hemoglobin.exposure')
+        self.threshold = data_values.MATERNAL_BMI.ANEMIA_THRESHOLD
+        p_low_anemic, p_low_non_anemic = self._sample_bmi_parameters(builder)
+        self.probability_low_given_anemic = builder.lookup.build_table(
+            p_low_anemic,
+        )
+        self.probability_low_given_non_anemic = builder.lookup.build_table(
+            p_low_non_anemic,
+        )
+
+        self.population_view = builder.population.get_view([
+            'pregnancy_status',
+            'maternal_bmi_propensity',
+        ])
+        builder.population.initializes_simulants(
+            self.on_initialize_simulants,
+            requires_streams=[self.name],
+            creates_columns=['maternal_bmi_propensity'],
+        )
+
+        self.maternal_bmi = builder.value.register_value_producer(
+            'maternal_bmi',
+            source=self.maternal_bmi,
+            requires_columns=['pregnancy_status', 'maternal_bmi_propensity'],
+        )
+
+    def on_initialize_simulants(self, pop_data: SimulantData):
+        propensity = self.randomness.get_draw(pop_data.index)
+        propensity = propensity.rename('maternal_bmi_propensity')
+        self.population_view.update(propensity)
+
+    def maternal_bmi(self, index: pd.Index):
+        pop = self.population_view.get(index)
+        p = pop['maternal_bmi_propensity']
+        p_low_anemic = self.probability_low_given_anemic(index)
+        p_low_non_anemic = self.probability_low_given_non_anemic(index)
+
+        pregnant = pop[pop['pregnancy_status'] == models.PREGNANT_STATE].index
+        pregnant_hemoglobin = self.hemoglobin(pregnant)
+        anemic = pregnant[pregnant_hemoglobin < self.threshold]
+        non_anemic = pregnant.difference(anemic)
```

Comment:
I want to make sure that "self.threshold" is equal to 100 g/L, which is the moderate threshold during pregnancy, not the any anemia threshold.

Also, I think we will want to save the assessment of anemia +/- 100g/L assessed at the start of pregnancy before it continues to change throughout pregnancy, attributing the cat1/cat2/cat3/cat4 exposure values so that we can easily write them out to the child model and assign the respective birthweight shifts.